stake-pool/program/src/instruction.rs

```diff
@@ -374,6 +379,48 @@ pub enum StakePoolInstruction {
     ///  11. `[]` Token program id
     ///  12. `[s]` (Optional) Stake pool sol withdraw authority
     WithdrawSol(u64),
+
+    /// Create token metadata for the stake-pool token in the
+    /// metaplex-token program
+    /// 0. `[]` Stake pool
+    /// 1. `[]` Manager
+    /// 2. `[]` Stake pool withdraw authority
+    /// 3. `[]` Pool token mint account
+    /// 4. `[]` Payer for creation of token metadata account
```

Comment:
That's a good question.  It may be because the specified `payer` is the same as the fee payer in your test transactions.  You can try a test with a different fee payer and see what happens!  I'm surprised that it lets this through though :thinking: